jenkins/image_building/initrd_sdk/unlock-mount-luks.sh

```diff
@@ -99,68 +159,300 @@ _dry_run() {
+        _luks_probe=$((_luks_probe + _result))
+        set -e
+        if [[ $_luks_probe -gt 0 ]]; then
+            # No reason to recheck if the modules can't be loaded they
+            # are most likely missing or incompatible
+            printf "INFO: dm_mod or dm_crypt modules can't be loaded!\n"
+            exit 1
+        fi
+        # Verify that the modules are loaded successfully
+        _crypt_lsmod=1
+        _dm_lsmod=1
+        while read -r _mod_name _ _ _; do
+            if [[ "${_mod_name}" == "dm_crypt" ]]; then
+                _crypt_lsmod=0
+            elif [[ "${_mod_name}" == "dm_mod" ]]; then
+                _dm_lsmod=0
+            fi
+        done < <(lsmod)
+        # If any of the models are seemingly not loaded that might be
+        # because lsmod is not displaying builtin modules so it has to
+        # be checked if a module is not loaded because it is builtin or
+        # because it can't load properly
+        if [[ ${_dm_lsmod} -eq 1 ]]; then
+            _dm_lsmod=$(_is_module_builtin "dm_mod")
+        fi
+        if [[ ${_crypt_lsmod} -eq 1 ]]; then
+            _crypt_lsmod=$(_is_module_builtin "dm_crypt")
+        fi
+        # Final checks
+        _ready=0
+        if [[ ${_crypt_lsmod} -eq 0 ]]; then
+            printf "INFO: dm_crypt check for LUKS : SUCCESS!\n"
+            _ready=$((++_ready))
+        else
+            printf "WARNING: dm_crypt check for LUKS : FAIL!\n"
+        fi
+        if [[ ${_dm_lsmod} -eq 0 ]]; then
+            printf "INFO: dm_mod for LUKS : SUCCESS!\n"
+            _ready=$((++_ready))
+        else
+            printf "WARNING: dm_mod check for LUKS : FAIL!\n"
+        fi
+        if [[ $_tpm_result -eq 0 ]]; then
+            printf "INFO: tpm2 persistent access to handles : SUCCESS!\n"
+            _ready=$((++_ready))
+        else
+            printf "WARNING: tpm2 persistent access to handles : FAIL!\n"
+        fi
+        if [[ ${_ready} -eq ${_exp_num_checks} ]]; then
+            printf "INFO: All checks are OK, proceeding with decryption!\n"
+            break
+        else
+            printf "WARNING: Some checks have FAILED, waiting 1s then retry!\n"
+        fi
+        _cnt=$((++_cnt))
+        _wait_sec 1
+    done
+    if [[ ! ${_ready} -eq ${_exp_num_checks} ]]; then
+        printf "ERROR: Some checks have FAILED, TIMEOUT has been reached!\n"
+        exit 1
+    fi
+}
+
+_get_kernel_param_value() {
+    # This function returns the value of a kernel command line parameter if
+    # the parameter exists and has separate value field.
+    # Arguments:
+    # - The "key/name" of the parameter that the function should be looking for
+    local _target_param _params
+    _target_param="${1:?}"
+
+    IFS=" " read -r -a _params < "/proc/cmdline"
+
+    for _param in "${_params[@]}"; do
+        _key="${_param%=*}"
+        _value="${_param#*=}"
+        if [[ "${_target_param}" == "${_key}" && "${_key}" != "${_value}" ]];\
+        then
```

Comment:
```suggestion
```

Reply:
It is like that because of the line length limit

Reply:
Just put it on a line of its own without `\`.